site/ja/tfx/guide/examplegen.md

```diff
@@ -0,0 +1,222 @@
+from tfx.components.example_gen.import_example_gen.component import ImportExampleGen
+
+examples = tfrecord_input(path_to_tfrecord_dir)
+example_gen = ImportExampleGen(input=examples)
+```
+
+## Span, Version, Split
+
+Span は訓練用の Example をグループ化したものです。データがファイルシステムに永続的に保存されている
+場合、それぞれの Span は別々のディレクトリに保存されているでしょう。Span が指し示す内容は TFX に
+ハードコードされていません。Span はある一日のデータを示すかもしれませんし、ある1時間のデータかも
+しれません。 Span は実行したいタスクにおいて有効な任意のグループのことを示します。
+
+それぞれの Span はデータの複数の Version を保持します。たとえば、ある Span に品質の良くないデータが
+含まれていて、そこからいくつかの Example を取り除くことを考えると、これは Span のバージョンをあげます。
+デフォルトでは TFX のコンポーネントは Span の最新のバージョンに対して操作を行います。
+
+Span の各バージョンは、さらに Split に分割される場合があります。もっとも典型的な例では、Span を
+訓練用と評価用の Split に分割します。
+
+![Spans and Splits](images/spans_splits.png)
+
+## カスタム input/output split
+
+Note: この機能は TFX 0.14 以降でのみ利用可能です。
+
+ExampleGen が出力する 学習/評価 データの比率を変更するためには、`output_config` を ExampleGen コンポーネントに設定してください。
+例を次に示します:
+
+```python
+from  tfx.proto import example_gen_pb2
+
+# Input has a single split 'input_dir/*'.
+# Output 2 splits: train:eval=3:1.
+output = example_gen_pb2.Output(
+             split_config=example_gen_pb2.SplitConfig(splits=[
+                 example_gen_pb2.SplitConfig.Split(name='train', hash_buckets=3),
+                 example_gen_pb2.SplitConfig.Split(name='eval', hash_buckets=1)
+             ]))
+examples = csv_input(input_dir)
+example_gen = CsvExampleGen(input=examples, output_config=output)
+```
+
+この例の中でどのように `hash_buckets` が設定されているかに注意してください。
+
+すでに入力データが分割されている場合、`input_config` を ExampleGen コンポーネントに設定してください。
+例を次に示します:
+
+```python
+from  tfx.proto import example_gen_pb2
+
+# Input train split is 'input_dir/train/*', eval split is 'input_dir/eval/*'.
+# Output splits are generated one-to-one mapping from input splits.
+input = example_gen_pb2.Input(splits=[
+                example_gen_pb2.Input.Split(name='train', pattern='train/*'),
+                example_gen_pb2.Input.Split(name='eval', pattern='eval/*')
+            ])
+examples = csv_input(input_dir)
+example_gen = CsvExampleGen(input=examples, input_config=input)
+```
+
+ファイルベースの ExampleGen コンポーネント (たとえば、 CsvExampleGen や ImportExampleGen) では、`pattern` は入力ファイルをまとめたディレクトリからの相対パスを glob 形式で記述したものになります。
+クエリベースの ExampleGen コンポーネント (たとえば、BigQUeryExampleGen や PrestoExampleGen) では、`pattern` はSQLクエリになります。
+
+デフォルトでは入力データをまとめたディレクトリには単一のファイルがあるものとして扱われます。また、学習/評価用のデータの分割は2:1の割合になるように行われます。
+
+詳細については [proto/example_gen.proto](https://github.com/tensorflow/tfx/blob/master/tfx/proto/example_gen.proto) を参照してください。
+
+### Span
+
+Note: この機能は TFX 0.15 以降でのみ利用可能です。
+
+Span は[入力する glob 形式の pattern](https://github.com/tensorflow/tfx/blob/master/tfx/proto/example_gen.proto) において '{SPAN}' を用いて指定することで利用できます:
+
+*   SPAN で指定した箇所は数字とマッチし、データを SPAN の番号に対応付けます。
+    たとえば、'data_{SPAN}-*.tfrecord' は 'data_12-a.tfrecord', 'date_12-b.tfrecord'
+    といったファイルを指し示します。
+*   SPAN が指定されない場合、常に Span を '0' 番として扱います。
+*   If SPAN is specified, pipeline will process the latest span, and store the
```

Comment:
[must] This line is already translated and can be removed.

Reply:
Oops, thanks!